src/main/java/bio/terra/service/iam/AuthenticatedUserRequest.java

```diff
@@ -55,7 +55,7 @@ public AuthenticatedUserRequest token(Optional<String> token) {
     @JsonIgnore
     public String getRequiredToken() {
         if (!token.isPresent()) {
-            throw new ApiException("Token required");
+            throw new IamUnauthorizedException("An OAuth token is required.");
```

Comment:
Maybe I don't fully get the flow of operation but if we're running behind an OIDC proxy and we get a request without a token, isn't that a 500? How does a request arrive without a token that isn't an internal server error?

Reply:
Well, in that case, the error isn't really _internal_, and from TDR's perpective, its an unauthorized request. The exception handling is a little weird in this area of the code. The two `ApiExceptions` in this and `SamRetry` are in different packages.